Refactored DocumentProcessingService and DocumentProcessingServiceTests

yav0 opened the pull request:
- **Code Architecture Improvements**:
    - Moved workflow-related methods (`WorkflowUsesTopics()`, `WorkflowUsesToc()`) to `Workflow` class
    - Created a dedicated service for workflow processing
    - Separated workflow service logic from document processing
    - Moved vector model error message to `EmbeddingsConnector`


- **Enhanced Test Coverage and Organization**:
    - Created dedicated `WorkflowServiceTests` class for better test isolation
    - Added test case verifying proper document replacement during workflow context reloading
    - Improved test setup with simplified document initialization
    - Fixed CSV overview assertion in the failing test

yav0 committed:
Refactor

artemkoloskov commented on Acme.Entities/Workflows/Workflow.cs:
```diff
@@ -39,4 +40,15 @@ public Workflow GetTopLevelWorkflow()
         return topLevelWorkflow;
     }
 
+    // Moved over from DocumentProcessingService to here as I think it makes more sense as a part of Workflow class
```
👍🏻

artemkoloskov commented on Acme.Services/DocumentProcessingService.cs:
```diff
@@ -14,20 +12,25 @@ namespace Acme.Services;
-        await DocumentAnalysisService.SetupPromptsAsync().ConfigureAwait(false);
-        await ExcelSheetAnalysisService.SetupPromptsAsync().ConfigureAwait(false);
+
+        await DocumentAnalysisService.SetupPromptsAsync();
```
Wonder, what is the reasoning behind removing the `.ConfigureAwait` calls?

yav0 replied:
I figured since there is no sync context to get back to, ConfigureAwait calls in the code served no real purpose and they could safely be removed for the sake of brevity. The continuation will just run on a thread pool thread, just as it would with ConfigureAwait(false).

artemkoloskov reviewed (changes requested):
Yavuz, thank you for the PR!

I've left a couple of comments, overall everything looks good code-vise.

Positives:
- It's clear you've got a good understanding of what this piece of code does
- The refactoring does make things look more logical, and more maintainable, very good
- There are no new warnings, especially i liked your attention to nullables!

Some things i would've like to see also:

- Descriptive PR description/commit messages
- More quantitative commits
- Some additional tests for newly created classes (even if just generated with LLMs)

I'd like to request a couple of changes, if you don't mind doing them:

- Try to address the negatives listed above
- You've mentioned that you noticed the broken test, and wanted to add some more tests but though that might be out of scope, which is a very valid reason. But let's assume you have more time, could you try to fix the broken test please? To clarify - this test was indeed failing for a long time in production because it was not a priority until we needed to fix it. So this is not a "gotcha", but it would be nice to see your approach to solving it.

yav0 committed:
Fix document processing unit test for csv overview assertion

yav0 committed:
Separate workflow service tests from document processing tests

• Extracted workflow tests into dedicated WorkflowServiceTests class
• Simplified test setup by removing DocumentProcessingService dependency
• Improved test isolation by manually initializing document pages
• Added more assertions to cover additional test cases
• Reduced test complexity by removing actual file processing dependencies

yav0 committed:
Add assertions to verify all document pages are included in workflow

yav0 committed:
Add test for clearing documents when workflow context reloaded

Added a new test case 'LoadContextToWorkflowAsync_WhenReloadingContext_ClearsExistingDocuments' that verifies the workflow service properly replaces existing documents when reloading context, ensuring old documents are removed and new ones are correctly added.
